pinot-common/src/main/java/org/apache/pinot/common/config/tuner/RealTimeAutoIndexTuner.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.config.tuner;
+
+import org.apache.pinot.spi.config.table.IndexingConfig;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.tuner.TableConfigTuner;
+import org.apache.pinot.spi.data.Schema;
+
+
+/**
+ * Used to auto-tune the table indexing config. It takes the original table
+ * config, table schema and adds the following to indexing config:
+ * - Inverted indices for all dimensions
+ * - No dictionary index for all metrics
+ */
+public class RealTimeAutoIndexTuner {
+
+  @TableConfigTuner(name = "realtimeAutoIndexTuner")
+  public static TableConfig tuneTableConfig(TableConfig initialConfig, Schema schema) {
+    IndexingConfig initialIndexingConfig = initialConfig.getIndexingConfig();
```

Comment:
Should think of adding one column as sorted. I am not sure how to pick that column.

Reply:
This is a very basic tuner. I will add more comprehensive ones in the follow up PRs.

Reply:
It will be great to look at the existing tuner (RecommenderDriver) and improve/finetune it. We implemented it for recommending indexes and other parts of the config.

Reply:
Sounds good. I'll follow up in subsequent PRs. Thanks for the suggestion !